bootstrap/interfaces/secret.go

```diff
@@ -23,6 +23,9 @@ type SecretProvider interface {
 	// Service key is use as the access token role which must have be previously setup.
 	GetAccessToken(tokenType string, serviceKey string) (string, error)
 
+	// ListSecretsAtPath returns a list of secret keys from an insecure/secure secrets secret store.
+	ListSecretsAtPath(path string) ([]string, error)
```

Comment:
This is not what the enhancement issue is requesting. Should be
```suggestion
	// ListSecretPaths returns a list of secret paths from the service's SecretStore.
	ListSecretPaths() ([]string, error)
```

Reply:
@lenny-intel Nothing in either the Consul nor Vault implementation limits the secret store to being a flat directory.  Both of them allow for additional hierarchy to be interposed between the prefix and the actual secret.

Reply:
Update: Lenny's comment here https://github.com/edgexfoundry/go-mod-bootstrap/issues/348#issuecomment-1239664562 directly addreses the above.  Basically, Lenny is advocating that we remove support for hierarchy.  I am ok with that.